spec/models/upload_certificate_event_spec.rb

```diff
@@ -293,8 +296,8 @@
     end
 
     it 'must reference an object' do
-      event = UploadCertificateEvent.create
-      expect(event.errors[:component]).to eql [t('components.errors.must_exist'), t('components.errors.must_exist')]
+      event = UploadCertificateEvent.create(component: bad_component)
+      expect(event.errors[:component]).to eql [t('components.errors.must_exist')]
     end
```

Comment:
I would still expect a test with three UploadCertificatesEvents where the last one is invalid and component ends up with 2 certs.